### ISSUE TITLE
Make sure C++ Interop works when cross-compiling with the new Swift 6 SDK bundle

### ISSUE THREAD

**Issue by finagolfin** [finagolfin/swift-android-sdk#183]
It works well with the new Android overlay when natively compiling, but cross-compilation hasn't been tested as much.

**Comment by lhoward** [comment]
Yes, couldn't find `swift/bridging` – not a biggie but might be useful for some future stuff.

**Comment by finagolfin** [comment]
OK, [that is only installed when the compiler is built](https://github.com/swiftlang/swift/blob/main/lib/ClangImporter/SwiftBridging/CMakeLists.txt), which this CI doesn't bother doing. I'll look into copying it over: try [the latest CI-built SDK bundles with libc++ support](https://github.com/finagolfin/swift-android-sdk/actions/runs/11680874837) and let me know if anything else is missing.

**Comment by finagolfin** [comment]
I made some changes and got a basic C++ interop test binary from the compiler validation suite to build. Try out [this 6.0.3 SDK bundle fresh out of the CI](https://github.com/finagolfin/swift-android-sdk/actions/runs/12300985934) and let me know if you see any issues.

**Comment by finagolfin** [comment]
@lhoward, fixed all known issues with interop: I'll close this when I upload the next patch release, unless you spot anything else.

**Comment by finagolfin** [comment]
The latest 6.0.3 patch release I just uploaded has some fixes for this, reopen if it's still failing for you.
